pkg/yamltemplate/filetests/ytt-library/overlay/when.yaml

```diff
@@ -0,0 +1,139 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:template", "template")
+
+#@ def test1_when_left():
+map:
+  input_string: val1
+  input_int: val1
+  input_func: val1
+  input_list: val1
+array:
+- val1
+array2:
+- val1
+array3:
+- val1
+#@ end
+
+#@ def true_func(num):
+#@ if num == 1:
+#@   return True
+#@ end
+#@ return False
+#@ end
+
+#@ def test1_when_right():
+map:
+  #@overlay/match when="1+"
+  #@overlay/replace
```

Comment:
we dont need all these replace btw.

Reply:
Wonderful. Removed them in the latest commit. Nice catch!